main.tf

```diff
@@ -1,236 +1,428 @@
-    ownership_verification_certificate_arn = var.domain_name_ownership_verification_certificate_arn
+    certificate_arn                        = local.create_certificate ? module.acm.acm_certificate_arn : var.domain_name_certificate_arn
     endpoint_type                          = "REGIONAL"
     security_policy                        = "TLS_1_2"
+    ownership_verification_certificate_arn = var.domain_name_ownership_verification_certificate_arn
   }
 
   dynamic "mutual_tls_authentication" {
-    for_each = length(keys(var.mutual_tls_authentication)) == 0 ? [] : [var.mutual_tls_authentication]
+    for_each = length(var.mutual_tls_authentication) > 0 ? [var.mutual_tls_authentication] : []
 
     content {
       truststore_uri     = mutual_tls_authentication.value.truststore_uri
       truststore_version = try(mutual_tls_authentication.value.truststore_version, null)
     }
   }
 
-  tags = merge(var.domain_name_tags, var.tags)
+  tags = var.tags
 }
 
-# Default stage log group
-resource "aws_cloudwatch_log_group" "this" {
-  count = var.create && var.create_default_stage && var.create_default_stage_access_log_group ? 1 : 0
+resource "aws_apigatewayv2_api_mapping" "this" {
+  count = local.create_domain_name && local.create_stage ? 1 : 0
+
+  api_id          = aws_apigatewayv2_api.this[0].id
+  api_mapping_key = var.api_mapping_key
+  domain_name     = aws_apigatewayv2_domain_name.this[0].id
+  stage           = aws_apigatewayv2_stage.this[0].id
+}
 
-  name              = coalesce(var.default_stage_access_log_group_name, "${var.name}${var.default_stage_access_log_group_name_suffix}")
-  retention_in_days = var.default_stage_access_log_group_retention_in_days
-  kms_key_id        = var.default_stage_access_log_group_kms_key_id
-  skip_destroy      = var.default_stage_access_log_group_skip_destroy
-  log_group_class   = var.default_stage_access_log_group_class
+################################################################################
+# Domain - Route53 Record
+################################################################################
 
-  tags = merge(var.tags, var.default_stage_access_log_group_tags)
+locals {
+  # https://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-custom-domains.html
+  stripped_domain_name = replace(var.domain_name, "*.", "")
 }
 
-# Default stage
-resource "aws_apigatewayv2_stage" "default" {
-  count = var.create && var.create_default_stage ? 1 : 0
+data "aws_route53_zone" "this" {
+  count = local.create_domain_name && var.create_domain_records ? 1 : 0
 
-  api_id      = aws_apigatewayv2_api.this[0].id
-  name        = "$default"
-  auto_deploy = true
+  name = local.stripped_domain_name
+}
 
-  dynamic "access_log_settings" {
-    for_each = (var.default_stage_access_log_destination_arn != null || var.create_default_stage_access_log_group) && var.default_stage_access_log_format != null ? [true] : []
+resource "aws_route53_record" "alias_ipv4" {
```

Comment:
No ipv6?

Reply:
I don't believe they support IPv6, but I can add the record - theres no harm (as far as I am aware) in having it even if IPv6 is not available

Reply:
added via variable to allow folks to tweak if necessary, but adds both `A` and `AAAA` records for IPv4 and IPc6 in https://github.com/terraform-aws-modules/terraform-aws-apigateway-v2/pull/54/commits/d6e694c0e73bf8c7fc7d145e074c3d562dc36848